### PR TITLE
[Google Blockly] Enable Sprite Lab Documentation with Google Blockly

### PR DESCRIPTION
**Pull request by mikeharv**
This includes a few updates that would add support for Sprite Lab Docs on Google Blockly. Specifically, it:

* Tells our documentation to use render Blocks with Google Blockly for the Sprite Lab environment index (/docs/ide/spritelab/) and each "expression" page linked on it.
* Adds a context menu option for blocks with `helpUrl` values that opens a modal documentation dialog. (Requires setting an experiment flag.)
* Updated `block_utils.js` to add `helpUrl`s to our custom blocks, based on existing `docFunc` values.

Because Sprite Lab is the only Blockly lab with docs, we can safely flip to Google Blockly in "all" labs. If, for example, Artist also had docs, we would need some conditional logic to choose between the `googleblockly.js` and `blockly.js` scripts.

### Example - `change costume to` block
This block is defined at `pools/GamelabJr/blocks/gamelab_setAnimation/edit`. 
The config JSON includes a `docFunc": "gamelab_setAnimation",` property.
image
(I have a local change that renders the above block with Google Blockly. Levelbuilder/staging are still using CDO Blockly.)

The documentation for this block can be viewed at `/docs/spritelab/gamelab_setAnimation`.

With the experiment enabled, a new "Get help with this block" option is shown in the context menu:
image

Selecting the option opens the modal, just like we do for Game Lab and App Lab:
image
(Note that the embedded project example shown in the documentation above uses CDO Blockly because the project actually lives on production.)

## Links

Jira: https://codedotorg.atlassian.net/browse/CT-168

## Follow-up work

We will also want to update the helpUrls of existing blocks. That work is handled here:
https://github.com/code-dot-org/code-dot-org/pull/55589

Beginning in April, the curriculum team will be working to update all block documentation as needed for the spring launch. The experiment flag will enable them to test everything as they write without risking that an external user might stumble upon any outdated docs.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/blockly/addons/contextMenu.js [hunk 1]
```diff
@@ -242,6 +244,52 @@ function registerThemes(themes) {
...
+        typeof block.helpUrl === 'function' ? block.helpUrl() : block.helpUrl;
+      // Some common Blockly blocks have help URLs for pages on Wikipedia, GitHub, etc.
+      // We only want to allow a documentation dialog for one of our local docs links.
+      if (url && url.startsWith('/docs/')) {
+        return 'enabled';
+      }
+      return 'hidden';
+    },
+    callback(scope) {
+      // scope.block.showHelp();
```
Comment (molly-moen) [inline apps/src/blockly/addons/contextMenu.js]: do we need to keep this commented out line?

File: dashboard/app/views/programming_environments/show.html.haml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 %link{href: asset_path('css/curriculum_navigation.css'), rel: 'stylesheet', type: 'text/css'}
-%script{src: webpack_asset_path('js/blockly.js')}
+%script{src: webpack_asset_path('js/googleblockly.js')}
```
Comment (molly-moen) [inline dashboard/app/views/programming_environments/show.html.haml:2]: why was this changed?

File: apps/src/blockly/addons/contextMenu.js (hunk 1 above)
Comment (mikeharv) [reply apps/src/blockly/addons/contextMenu.js]: Nope! That was just what this menu option does in core Blockly. Will remove.

File: dashboard/app/views/programming_environments/show.html.haml (hunk 2 above)
Comment (mikeharv) [reply dashboard/app/views/programming_environments/show.html.haml:2]: This is what tells our documentation to start rendering blocks with Google Blockly instead of CDO Blockly.

File: dashboard/app/views/programming_expressions/show.html.haml [hunk 3]
```diff
@@ -1,9 +1,9 @@
-//%script{src: webpack_asset_path('js/googleblockly.js')}
+%script{src: webpack_asset_path('js/googleblockly.js')}
 %link{href: asset_path('css/markdown.css'), rel: 'stylesheet', type: 'text/css'}
 %link{href: asset_path('css/curriculum_navigation.css'), rel: 'stylesheet', type: 'text/css'}
 - content_for(:head) do
   = tag 'meta', name: 'description', content: @programming_expression.syntax
-%script{src: webpack_asset_path('js/blockly.js')}
+//%script{src: webpack_asset_path('js/googleblockly.js')}
```
Comment (molly-moen) [inline dashboard/app/views/programming_expressions/show.html.haml]: remove this?

**Review by molly-moen — APPROVED** [review]

File: dashboard/app/views/programming_expressions/show.html.haml (hunk 3 above)
Comment (mikeharv) [reply dashboard/app/views/programming_expressions/show.html.haml]: Yes, thank you! Removed.
